### PR TITLE
Added support for PMA_PASSWORD_FILE

### PR DESCRIPTION
**Pull request by Mandy91**
Documentation said there is support for, but there was no code to implement this

### CODE REVIEW

**Comment by williamdes** [comment]
@davidmwhynot is this change needed ?
#232

**Comment by davidmwhynot** [comment]
Based on [this](https://github.com/phpmyadmin/docker/issues/108#issuecomment-519323193) I would say no.

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -47,4 +47,8 @@ EOT
     fi
 fi
 
+if [ -n "$PMA_PASSWORD_FILE" ]; then
+	export PMA_PASSWORD="$(cat $PMA_PASSWORD_FILE)"
```
Comment (davidmwhynot) [inline docker-entrypoint.sh]: This implementation bypasses one of the main advantages of using docker secrets: sensitive information is not stored directly in the an environment variable. Instead, it is read from a secret file at runtime by application code. Nothing sensitive should ever be stored directly in an environment variable, as is being done here.

**Comment by davidmwhynot** [comment]
> Based on [this](https://github.com/phpmyadmin/docker/issues/108#issuecomment-519323193) I would say no.

After doing some digging through the repository, I am unable to find an implementation for actually reading secrets from a file path specified by an environment variable. So not really sure where [this](https://github.com/phpmyadmin/docker/issues/108#issuecomment-519323193) information came from.

**Comment by williamdes** [comment]
@J0WI can I please have your review on this one ?

**Comment by J0WI** [comment]
Not sure why #232 have been merged, this functionality was never implemented.
@Mandy91 there is a great and roust implementation in #206 (taken from [here](https://github.com/docker-library/wordpress/blob/master/docker-entrypoint.sh#L4-L24)). Would you like to pick this up again?

**Comment by williamdes** [comment]
> Not sure why #232 have been merged, this functionality was never implemented.

I trusted the comment and then merged the PR..

**Comment by Mandy91** [comment]
@J0WI Sure no problem. Would the following environment variables suffice to implement the password file for? To me it doesn't make much sense to add it for the username environment variable etc.
- MYSQL_ROOT_PASSWORD
- MYSQL_PASSWORD
- PMA_PASSWORD

**Comment by J0WI** [comment]
> To me it doesn't make much sense to add it for the username environment variable etc.

Sure
cc @williamdes

**Comment by williamdes** [comment]
I agree

File: testing/conftest.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+def password(request):
+    return request.config.getoption("--password")
+
+@pytest.fixture
+def server(request):
+    return request.config.getoption("--server")
+    
+@pytest.fixture
+def sqlfile(request):
+    return request.config.getoption("--sqlfile")
```
Comment (williamdes) [inline testing/conftest.py]: can you add a final line ?

File: testing/phpmyadmin_test.py [hunk 3]
```diff
@@ -20,12 +20,14 @@ def test_phpmyadmin(url, username, password, server=None, sqlfile=None):
...
         else:
-            sqlfile = './testing/world.sql'
+            path = os.path.dirname(os.path.realpath(__file__))
+            sqlfile = path + '/world.sql'
     br = mechanize.Browser()
 
     # Ignore robots.txt
     br.set_handle_robots(False)
 
+
```
Comment (williamdes) [inline testing/phpmyadmin_test.py]: can you remove the new line here please

**Comment by Mandy91** [comment]
@williamdes Can we somehow trigger Travis to see if the tests which I have created work properly?

File: docker-entrypoint.sh [hunk 4]
```diff
@@ -47,4 +47,17 @@ EOT
     fi
 fi
 
+function get_docker_secret() {
+    local env_var="${1}"
+    local env_var_file="${env_var}_FILE"
+
+    if [[ -n "${!env_var_file}" ]]; then
```
Comment (williamdes) [inline docker-entrypoint.sh]: can you add a comment because it is unclear what this line does

File: testing/run-test-locally.sh [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    exit 99
+}
+
+trap restore 0 1 2 3 6 14 15
+
+docker kill mariadb
+docker kill phpmyadmin
+
+set -e
+docker run --rm -d -p 3300:3306 -e MYSQL_ROOT_PASSWORD=my-secret-pw --init --hostname mariadb --name mariadb --network phpmyadmin --tmpfs /var/lib/mysql:rw,noexec,nosuid,size=300m mariadb:10.3
```
Comment (williamdes) [inline testing/run-test-locally.sh]: coud you use some ENV variables that have default values like `${DOCKER_MYSQL_PORT:-3306}` ?

**Comment by williamdes** [comment]
> @williamdes Can we somehow trigger Travis to see if the tests which I have created work properly?

I asked @ibennetch

**Comment by williamdes** [comment]
Thank you for your work ! :100:

File: docker-compose.testing.yml [hunk 6]
```diff
@@ -1,26 +1,27 @@
...
   db_server:
-    image: ${DB}
+    image: ${DB:-mariadb:10.3}
     container_name: phpmyadmin_testing_db
     environment:
-      - MYSQL_ROOT_PASSWORD=${TESTSUITE_PASSWORD}
+      - MYSQL_ROOT_PASSWORD=${TESTSUITE_PASSWORD:-my-secret-pw}
     volumes:
       - ./testing/testing.cnf:/etc/mysql/conf.d/testing.cnf:ro
+    tmpfs:
```
Comment (williamdes) [inline docker-compose.testing.yml]: will have issues on MacOS but I agree to have that line

**Comment by williamdes** [comment]
@Mandy91 did you discover why we have randomly failing E2E tests here ?

**Comment by Mandy91** [comment]
> @Mandy91 did you discover why we have randomly failing E2E tests here ?

Yes, initially I removed the variable FILENAME in test-docker.sh, I added it back and now it is working again :+1:

**Comment by eriksenonline** [comment]
Is it possible to add support for PMA_HOSTS_FILE and PMA_HOST_FILE to ?

**Comment by williamdes** [comment]
@Mandy91 I rebased your work

**Comment by williamdes** [comment]
@Mandy91 I rebased and squashed your work

**Comment by williamdes** [comment]
@Mandy91 If you want to edit your commit to use your account like previous ones please do so because I will merge in less than one hour :)

**Comment by williamdes** [comment]
`From: Mandy <mandy@Mandys-MacBook-Pro.local>`

**Comment by williamdes** [comment]
Merged, thank you for your contribution @Mandy91 !
You added tests and I love it !

I added b295aa6bf722a5e95d4b4b185e367e9a2d9c5916 after merging

**Pull request merged**